packages/react/src/internal/deprecate.js

```diff
@@ -26,8 +27,37 @@ export default function deprecate(propType, message) {
       );
     }
 
-    return propType(props, propName, componentName, ...rest); // eslint-disable-line consistent-return
+    return propType(props, propName, componentName, ...rest);
```

Comment:
This change and the `useMerge` helper are nice. Should we bring them into next as well in order to use them until the major transition.

Reply:
Sure, I can easily cherry-pick them into next.